web3/main.py

```diff
@@ -258,6 +259,18 @@ def isConnected(self) -> bool:
     def is_encodable(self, _type: TypeStr, value: Any) -> bool:
         return self.codec.is_encodable(_type, value)
 
+    @property
+    def codec(self) -> ABICodec:
+        return self._codec
+
+    @codec.getter
+    def codec(self) -> ABICodec:
```

Comment:
I don't follow the need for these property methods.  Why not just assign `self.codec = ABICodec(...)`?

Reply:
oops, forgot to take this out, thanks!